model/nh_utils.F90

```diff
@@ -195,7 +194,7 @@ subroutine update_dz_c(is, ie, js, je, km, ng, dt, dp0, zs, area, ut, vt, gz, ws
 6000 continue
 
 ! Enforce monotonicity of height to prevent blowup
-!$OMP parallel do default(none) shared(is1,ie1,js1,je1,ws,zs,gz,rdt,km)
+!$OMP parallel do default(none) shared(is1,ie1,js1,je1,ws,zs,gz,rdt,km,dz_min)
   do j=js1, je1
      do k=2, km+1
```

Comment:
This is different from current EMC fork, where I think Kate switched back to k=km,1,-1 in her [PR#53](https://github.com/NOAA-EMC/GFDL_atmos_cubed_sphere/pull/53)
https://github.com/NOAA-EMC/GFDL_atmos_cubed_sphere/pull/53/files#diff-ab35d906facb07c24877b72bbc6b6acc3e856a82cb065a5a6d3391f3a149969aR200

Reply:
After discussions with @junwang-noaa, I have included the master version of the height monotonicity constraint for update_dz_*

Reply:
@junwang-noaa can you take one more look and approve if this is now good - thanks!